python/src/kagent/a2a.py

```diff
@@ -0,0 +1,115 @@
+#! /usr/bin/env python3
+
+import logging
+import os
+from contextlib import asynccontextmanager
+
+import httpx
+from a2a.auth.user import User
+from a2a.server.agent_execution import RequestContext, SimpleRequestContextBuilder
+from a2a.server.apps import A2AStarletteApplication
+from a2a.server.context import ServerCallContext
+from a2a.server.request_handlers import DefaultRequestHandler
+from a2a.server.tasks import TaskStore
+from a2a.types import AgentCard, MessageSendParams, Task
+from fastapi import FastAPI, Request
+from fastapi.responses import PlainTextResponse
+from google.adk.a2a.executor.a2a_agent_executor import A2aAgentExecutor
+from google.adk.agents import BaseAgent
+from google.adk.runners import Runner
+
+from kagent import KAgentSessionService, KAgentTaskStore
+
+# --- Constants ---
+USER_ID = "admin@kagent.dev"
+
+# --- Configure Logging ---
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class KAgentUser(User):
+    def __init__(self, user_id: str):
+        self.user_id = user_id
+
+    @property
+    def is_authenticated(self) -> bool:
+        return False
+
+    @property
+    def user_name(self) -> str:
+        return self.user_id
+
+
+class KAgentRequestContextBuilder(SimpleRequestContextBuilder):
+    """
+    A request context builder that will be used to hack in the user_id for now.
+    """
+
+    def __init__(self, user_id: str, task_store: TaskStore):
+        super().__init__(task_store=task_store)
+        self.user_id = user_id
+
+    async def build(
+        self,
+        params: MessageSendParams | None = None,
+        task_id: str | None = None,
+        context_id: str | None = None,
+        task: Task | None = None,
+        context: ServerCallContext | None = None,
+    ) -> RequestContext:
+        if not context:
```

Comment:
i assume in the future we will want to grab the user from the session / request. is that going to be possible?

Reply:
Yea it will definitely be possible, A2A has support for lots of auth schemes, and we can also add our own if we need